### ISSUE TITLE
Check for Backward Incompatibilities Between MPI 1.0 and MPI 1.1

### ISSUE THREAD

**Issue by wesbland** [mpi-forum/mpi-issues#415]
# Problem

- [ ] Mention that the MPI 1.1 specification is backward incompatible with MPI 1.0.
- [ ] Look to see if we can enumerate the list of incompatibilities from MPI 1.0 to MPI 1.1.

# Proposal

# Changes to the Text

# Impact on Implementations

# Impact on Users

# References

**Comment by hritzdorf** [comment]
I think that the addition of MPI_ERROR to MPI_Status was not backward compatible. It increased the size of MPI_Status objects. Thus, problems in functions returning MPI_Status and functions with arrays of statuses.

**Comment by wgropp** [comment]
That depends on your definition of backward incompatibilities.  When talking about a standard, backward compatibility does *not* mean binary compatible - that's not a standard-relevant definition. I'd ask if an MPI 1.0 program, as source code, compiled with the MPI 1.1 definition, would function correctly. Since MPI 1.0 provided MPI_STATUS_SIZE for Fortran and didn't define all of the elements of MPI_Status in C, I don't see this as a backward incompatibility. No *correct* MPI 1.0 program would fail to compile and function correctly with an MPI 1.1 implementation because of an addition to MPI_Status.

**Comment by wesbland** [comment]
I agree with @wgropp that `MPI_ERROR` is not an issue with the standard. Many implementations aren't even binary compatible with themselves over time. I believe OMPI makes no ABI guarantees across major versions. MPICH has had its own AI compatibility initiative for a while, but it doesn't go back to MPI 1.0.

Unfortunately, the lack of a CHANGELOG for MPI 1.1 makes finding any differences here impractical.
